File: src/extras-kio/libkpasswdserverclient.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KIO__AuthInfo
#include <KPasswdServerClient>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <kpasswdserverclient.h>
#include "libkpasswdserverclient.h"
#include "libkpasswdserverclient.hxx"

KPasswdServerClient* KPasswdServerClient_new() {
    return new KPasswdServerClient();
}

bool KPasswdServerClient_CheckAuthInfo(KPasswdServerClient* self, KIO__AuthInfo* info, long long windowId, long long usertime) {
    return self->checkAuthInfo(info, static_cast<qlonglong>(windowId), static_cast<qlonglong>(usertime));
}

int KPasswdServerClient_QueryAuthInfo(KPasswdServerClient* self, KIO__AuthInfo* info, const libqt_string errorMsg, long long windowId, long long usertime) {
    QString errorMsg_QString = QString::fromUtf8(errorMsg.data, errorMsg.len);
    return self->queryAuthInfo(info, errorMsg_QString, static_cast<qlonglong>(windowId), static_cast<qlonglong>(usertime));
}

void KPasswdServerClient_AddAuthInfo(KPasswdServerClient* self, const KIO__AuthInfo* info, long long windowId) {
    self->addAuthInfo(*info, static_cast<qlonglong>(windowId));
}

void KPasswdServerClient_RemoveAuthInfo(KPasswdServerClient* self, const libqt_string host, const libqt_string protocol, const libqt_string user) {
    QString host_QString = QString::fromUtf8(host.data, host.len);
    QString protocol_QString = QString::fromUtf8(protocol.data, protocol.len);
    QString user_QString = QString::fromUtf8(user.data, user.len);
    self->removeAuthInfo(host_QString, protocol_QString, user_QString);
}

void KPasswdServerClient_Delete(KPasswdServerClient* self) {
    delete self;
}

File: include_original/extras-kio/libkpasswdserverclient.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// https://api.kde.org/kpasswdserverclient.html
pub const kpasswdserverclient = struct {
    /// New constructs a new KPasswdServerClient object.
    ///
    ///
    pub fn New() QtC.KPasswdServerClient {
        return qtc.KPasswdServerClient_new();
    }

    /// [Qt documentation](https://api.kde.org/kpasswdserverclient.html#checkAuthInfo)
    ///
    /// ``` self: QtC.KPasswdServerClient, info: QtC.KIO__AuthInfo, windowId: i64, usertime: i64 ```
    pub fn CheckAuthInfo(self: ?*anyopaque, info: ?*anyopaque, windowId: i64, usertime: i64) bool {
        return qtc.KPasswdServerClient_CheckAuthInfo(@ptrCast(self), @ptrCast(info), @intCast(windowId), @intCast(usertime));
    }

    /// [Qt documentation](https://api.kde.org/kpasswdserverclient.html#queryAuthInfo)
    ///
    /// ``` self: QtC.KPasswdServerClient, info: QtC.KIO__AuthInfo, errorMsg: []const u8, windowId: i64, usertime: i64 ```
    pub fn QueryAuthInfo(self: ?*anyopaque, info: ?*anyopaque, errorMsg: []const u8, windowId: i64, usertime: i64) i32 {
        const errorMsg_str = qtc.libqt_string{
            .len = errorMsg.len,
            .data = errorMsg.ptr,
        };
        return qtc.KPasswdServerClient_QueryAuthInfo(@ptrCast(self), @ptrCast(info), errorMsg_str, @intCast(windowId), @intCast(usertime));
    }

    /// [Qt documentation](https://api.kde.org/kpasswdserverclient.html#addAuthInfo)
    ///
    /// ``` self: QtC.KPasswdServerClient, info: QtC.KIO__AuthInfo, windowId: i64 ```
    pub fn AddAuthInfo(self: ?*anyopaque, info: ?*anyopaque, windowId: i64) void {
        qtc.KPasswdServerClient_AddAuthInfo(@ptrCast(self), @ptrCast(info), @intCast(windowId));
    }

    /// [Qt documentation](https://api.kde.org/kpasswdserverclient.html#removeAuthInfo)
    ///
    /// ``` self: QtC.KPasswdServerClient, host: []const u8, protocol: []const u8, user: []const u8 ```
    pub fn RemoveAuthInfo(self: ?*anyopaque, host: []const u8, protocol: []const u8, user: []const u8) void {
        const host_str = qtc.libqt_string{
            .len = host.len,
            .data = host.ptr,
        };
        const protocol_str = qtc.libqt_string{
            .len = protocol.len,
            .data = protocol.ptr,
        };
        const user_str = qtc.libqt_string{
            .len = user.len,
            .data = user.ptr,
        };
        qtc.KPasswdServerClient_RemoveAuthInfo(@ptrCast(self), host_str, protocol_str, user_str);
    }

    /// [Qt documentation](https://api.kde.org/kpasswdserverclient.html#dtor.KPasswdServerClient)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KPasswdServerClient ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KPasswdServerClient_Delete(@ptrCast(self));
    }
};
